src/main/java/graphql/execution/ExecutionStepInfo.java

```diff
@@ -30,19 +31,37 @@
     private final ExecutionPath path;
     private final ExecutionStepInfo parent;
 
-    // field, fieldDefinition and arguments stay the same for steps inside a list field
+    // field, fieldDefinition, fieldContainer and arguments stay the same for steps inside a list field
     private final MergedField field;
     private final GraphQLFieldDefinition fieldDefinition;
+    private final GraphQLObjectType fieldContainer;
     private final Map<String, Object> arguments;
 
-    private ExecutionStepInfo(GraphQLOutputType type, GraphQLFieldDefinition fieldDefinition, MergedField field, ExecutionPath path, ExecutionStepInfo parent, Map<String, Object> arguments) {
+    private ExecutionStepInfo(GraphQLOutputType type,
+                              GraphQLFieldDefinition fieldDefinition,
+                              MergedField field,
+                              ExecutionPath path,
+                              ExecutionStepInfo parent,
+                              Map<String, Object> arguments,
+                              GraphQLObjectType fieldsContainer) {
         this.fieldDefinition = fieldDefinition;
         this.field = field;
         this.path = path;
         this.parent = parent;
         this.type = assertNotNull(type, "you must provide a graphql type");
         this.arguments = arguments;
+        this.fieldContainer = fieldsContainer;
+    }
 
+    /**
+     * The GraphQLObjectType where fieldDefinition is defined.
+     * Note:
+     * For the Introspection field __typename the returned object type doesn't actually contain the fieldDefinition.
+     *
+     * @return GraphQLObjectType defining {@link #getFieldDefinition()}
+     */
+    public GraphQLObjectType getFieldContainer() {
+        return fieldContainer;
     }
 
```

Comment:
Why not use graphql.schema.GraphQLFieldsContainer as the type and not the more generic ObjectType.

This is what this class is for, type safe access to the fields

Reply:
ignore me - its resolved as an object type